src/Authenticator/LoginHandler.php

```diff
@@ -233,14 +249,20 @@ public function finishRegistration(HTTPRequest $request)
             );
         }
 
-        // Create the RegisteredMethod object
-        $registeredMethod = RegisteredMethod::create([
-            'MethodClassName' => get_class($method),
-            'Data' => json_encode($data),
-        ]);
+        if (!empty($data)) {
+            $member = $sessionStore->getMember();
+
+            // Fetch an existing RegisteredMethod object from the Member or make a new one
+            $registeredMethod =
+                $this->getMethodRegistry()->getRegisteredMethodFromMember($member, $method->getURLSegment())
+                    ?: RegisteredMethod::create(['MethodClassName' => get_class($method)]);
+
+            $registeredMethod->Data = json_encode($data);
+            $registeredMethod->write();
 
-        // Add it to the member
-        $sessionStore->getMember()->RegisteredMFAMethods()->add($registeredMethod);
+            // Add it to the member
+            $member->RegisteredMFAMethods()->add($registeredMethod);
```

Comment:
This whole block feels like it could be in a service class or on the Member extension

Reply:
Yeah I agree. I'm just wondering how that would look. I prefer not putting too much business logic in models so I'm keen on a service class. Maybe there should be one for all interaction between members and methods? The `MethodRegistry` already has a `getRegisteredMethodFromMember` which feels a little out of place.

Any suggestions?

Reply:
`RegisteredMethodLocatorService` ?